src/gui/qgscompoundcolorwidget.cpp

```diff
@@ -719,6 +743,12 @@ void QgsCompoundColorWidget::setColor( const QColor &color )
     //opacity disallowed, so don't permit transparent colors
     fixedColor.setAlpha( 255 );
   }
+
+  if ( mColorModel->currentIndex() && fixedColor.spec() != QColor::Cmyk )
```

Comment:
This is fragile if we reorder the combo box -- see the comment in the ui file regarding using user data here instead.

Reply:
Done